addons/event/views/event_registration_views.xml

```diff
@@ -247,11 +247,32 @@
                                 invisible="context.get('registration_view_hide_group_by_create_date_week')"/>
                         <filter string="Registration Date" name="group_by_create_date_day" domain="[]" context="{'group_by': 'create_date:day'}"
                                 invisible="not context.get('registration_view_hide_group_by_create_date_week')"/>
+                        <filter string="Campaign" name="group_by_utm_campaign_id" domain="[]"
+                                context="{'group_by': 'utm_campaign_id'}"/>
+                        <filter string="Medium" name="group_by_utm_medium_id" domain="[]"
+                                context="{'group_by': 'utm_medium_id'}"/>
+                        <filter string="Source" name="group_by_utm_source_id" domain="[]"
+                                context="{'group_by': 'utm_source_id'}"/>
                    </group>
                 </search>
             </field>
         </record>
 
+        <!-- Search across multiple events, add event search (name, organizer, ...) -->
+        <record id="view_registration_search_not_specific_to_an_event" model="ir.ui.view">
```

Comment:
If you define another view, you have to give priorities to clearly state which one is the default one. Odoo takes the one with the lowest priority as default.